platform/mbed_assert.h

```diff
@@ -77,61 +77,40 @@ do {                                                     \
  *  The assertion acts as a declaration that can be placed at file scope, in a
  *  code block (except after a label), or as a member of a C++ class/struct/union.
  *
- *  @note
- *  Use of MBED_STATIC_ASSERT as a member of a struct/union is limited:
- *  - In C++, MBED_STATIC_ASSERT is valid in class/struct/union scope.
- *  - In C, MBED_STATIC_ASSERT is not valid in struct/union scope, and
- *    MBED_STRUCT_STATIC_ASSERT is provided as an alternative that is valid
- *    in C and C++ class/struct/union scope.
- *
  *  @code
- *  MBED_STATIC_ASSERT(MBED_LIBRARY_VERSION >= 120,
- *          "The mbed library must be at least version 120");
+ *  MBED_STATIC_ASSERT(MBED_MAJOR_VERSION  >= 6,
+ *          "The mbed-os library must be at least version 6.0.0");
  *
  *  int main() {
  *      MBED_STATIC_ASSERT(sizeof(int) >= sizeof(char),
  *              "An int must be larger than a char");
  *  }
  *  @endcode
  */
-#if defined(__cplusplus) && (__cplusplus >= 201103L || __cpp_static_assert >= 200410L)
+#if defined(__cplusplus) && __cplusplus >= 201402L
```

Comment:
The C++ and C version tests aren't required. Code base is permitted to just assume C++14 and C11.

Plenty of C++14 assumptions in-tree already. Although maybe not C11. But GNU C, ARMC6 and IAR are all set to C11.

Reply:
Ok. I will further simplify this.